### DISCUSSION TITLE
Performance issues with k8s metrics collecting

### DISCUSSION THREAD

**Discussion by SergBw** [apache/skywalking#11860] [Q&A] [answered]
Hi!
I am trying to use Skywaking for monitoring my k8s cluster. I use the 9.7 version that was installed by the 4.5.0 helm chart. I configured metrics collecting by the docs (https://skywalking.apache.org/docs/main/v9.7.0/en/setup/backend/backend-k8s-monitoring/) and used the same configs and versions of agents as in the showcase repository.
The 'kubernetes-cadvisor' scraping rule works fine but enabling the "kube-state-metrics" rule causes performance issues. CPU usage of OAP grows from 0.5 to 4.5 vCPU. Opentelemetry agent starts posting errors like "Exporting failed. Will retry the request after interval" and its memory usage grows to infinity. After a few retries, it drops collected metrics so they don't reach skywalking and ui dashboards are empty. At the same time elasticsarch db (that is used as storage) uses only 10% CPU and 60% of memory of 6cpu/12 Gb mem machine.
I tried to increase OAP pods number to 5 or switch to BanyanDB but it didn't help.
Cluster size is 6-8 nodes and ~140 pods.
Will be grateful for any ideas about performance increasing!

**Comment by wu-sheng** [comment]
I am not sure what is your case, but monitor k8s seems not to need to have so many OAP pods.

**Comment by SergBw** [comment, marked as answer]
Hi. I have found that Skywalking needs a lot of CPU for proper working. My configuration requires 5-7 at the start and 1.5-3 at idle. 
So my solution is to deploy only 1 Skywalking pod at nodes that have 8+ CPU and set requests and limits in Helm chart to 8 CPU
